Main.hs

```diff
@@ -4,22 +4,47 @@ import Control.Monad (unless)
 import Text.Printf (printf)
 
 short :: [a] -> Bool
-short = undefined 
+short [] = True
+short [_] = True
+short [_, _] = True 
+short _ = False 
 
 lovely :: [Int] -> Bool
-lovely = undefined 
+lovely xs = short xs || xs !! 2 == 14
 
 rightTriangles :: [(Int, Int, Int)]
-rightTriangles = undefined 
+rightTriangles = [(a, b, c) | c <- [1..], b <- [1..c], a <- [1..b], a^2 + b^2 == c^2] 
 
 fizzBuzz :: [String]
-fizzBuzz = undefined 
+fizzBuzz = [fizzBuzz x | x <- [1..]]
+  where
+    fizzBuzz n
+      | n `mod` 15 == 0 = "FizzBuzz"
+      | n `mod` 3 == 0  = "Fizz"
+      | n `mod` 5 == 0  = "Buzz"
+      | otherwise = show n 
 
 ageOn :: String -> Float -> Float
-ageOn planet ageInSeconds = undefined 
+ageOn planet ageInSeconds = case planet of
+    "Mercury" -> (ageInSeconds / (0.2408467 * earthYearInSeconds))
+    "Venus" -> (ageInSeconds / (0.61519726 * earthYearInSeconds))
+    "Earth" -> (ageInSeconds / earthYearInSeconds)
+    "Mars" -> (ageInSeconds / (1.8808158 * earthYearInSeconds))
+    "Jupiter" -> (ageInSeconds / (11.862615 * earthYearInSeconds))
+    "Saturn" -> (ageInSeconds / (29.447498 * earthYearInSeconds))
+    "Uranus"  -> (ageInSeconds / (84.016846 * earthYearInSeconds))
+    "Neptune" -> (ageInSeconds / (164.79132 * earthYearInSeconds))
```

Comment:
It's better to avoid code duplication by introducing a function that does this computation.